src/us/kbase/workspace/database/ObjectInformation.java

```diff
@@ -301,71 +206,206 @@ public boolean equals(Object obj) {
+		 * @return this builder for chaining.
+		 */
+		public Builder withVersion(final int version) {
+			this.version = gt0(version, "version");
+			return this;
 		}
-		if (version != other.version) {
-			return false;
+		
+		/** Adds the required user that saved or copied the object to the builder.
+		 * @param user the user.
+		 * @return this builder for chaining.
+		 */
+		public Builder withSavedBy(final WorkspaceUser user) {
+			this.savedBy = requireNonNull(user, "user");
+			return this;
 		}
-		if (workspaceID != other.workspaceID) {
-			return false;
+		
+		/** Adds the required workspace information for the workspace containing the object
+		 * to the builder.
+		 * @param workspace the workspace ID.
+		 * @return this builder for chaining.
+		 */
+		public Builder withWorkspace(final ResolvedWorkspaceID workspace) {
+			this.ws = requireNonNull(workspace, "workspace");
+			return this;
+		}
+		
+		/** Add the required md5 checksum of the object when serialized to a JSON string with
+		 * sorted keys to the builder.
+		 * @param checksum the checksum
+		 * @return this builder for chaining.
+		 */
+		public Builder withChecksum(final MD5 checksum) {
+			this.chksum = requireNonNull(checksum, "checksum").getMD5();
+			return this;
+		}
+		
+		/** Add the required md5 checksum of the object when serialized to a JSON string with
+		 * sorted keys to the builder.
+		 * @param checksum the checksum
+		 * @return this builder for chaining.
+		 */
+		public Builder withChecksum(final String checksum) {
+			// again could do more checking here but should be built internally only so meh
+			this.chksum = checkString(checksum, "checksum");
+			return this;
+		}
+
+		/** Add the required size of the object when serialized to a JSON string to the builer.
+		 * @param size the object size.
+		 * @return this builder for chaining.
+		 */
+		public Builder withSize(final long size) {
+			this.size = gt0(size, "size");
+			return this;
+		}
+
+		/** Add the optional user provided and / or autogenerated metadata to the builder.
+		 * 
+		 * A null argument will remove any previously set metadata.
+		 * @param metadata the metadata
+		 * @return this builder for chaining.
+		 */
+		public Builder withUserMetadata(final UncheckedUserMetadata metadata) {
+			this.usermeta = metadata;
+			return this;
 		}
-		if (workspaceName == null) {
-			if (other.workspaceName != null) {
-				return false;
+		
+		/** Create the {@link ObjectInformation}. All fields are required other than the metadata.
+		 * @return the new {@link ObjectInformation}.
+		 */
+		public ObjectInformation build() {
+			if (	objectId < 1 ||
+					objectName == null ||
+					type == null ||
+					savedDate == null ||
+					version < 1 ||
+					savedBy == null ||
+					ws == null ||
+					chksum == null ||
+					size < 1) {
+				throw new IllegalArgumentException("One or more of the required arguments are "
+						+ "not set. Please check the documentation for the builder.");
```

Comment:
If you want to be extra accurate in your error messaging...

"One or more of the required arguments are invalid or not set."

(objectId / size / version are set, they just don't pass the tests)